Buggy behavior when deployed on ECS

### Describe the bug a clear and concise description of what the bug is.

When running this prometheus exporter as an ECS task, the following behavior has been observed by our customers: 

- Randomly getting 503's in the CloudWatch logs for the exporter service - prometheus-prefect-exporter - [ERROR] 503 Server Error: Service Unavailable for url
- With a scrape period of 10s, flow state changes are sometimes missed or duplicated. There is no way to tell what state the flow run is in without going to the Prefect Cloud UI

PromQL
```
max by (flow_name) (

6 * (max_over_time(prefect_info_flow_runs_total{cluster_arn=~"$ecs_cluster_arn",container_name!~"aws-guardduty-agent.*", state_name="Cancelled",}[$__interval]))
or  5 * (max_over_time(prefect_info_flow_runs_total{cluster_arn=~"$ecs_cluster_arn",container_name!~"aws-guardduty-agent.*",state_name="Crashed"}[$__interval]))
or 4 * (max_over_time(prefect_info_flow_runs_total{cluster_arn=~"$ecs_cluster_arn",container_name!~"aws-guardduty-agent.*",state_name="Failed"}[$__interval]))
or 3 * max_over_time(prefect_info_flow_runs_total{cluster_arn=~"$ecs_cluster_arn",container_name!~"aws-guardduty-agent.*",state_name="Running"}[$__interval])
or 2 * (max_over_time(prefect_info_flow_runs_total{cluster_arn=~"$ecs_cluster_arn",container_name!~"aws-guardduty-agent.*",state_name="Pending"}[$__interval]))
or   7*(  max_over_time(prefect_info_flow_runs_total{cluster_arn=~"$ecs_cluster_arn",container_name!~"aws-guardduty-agent.*",state_name="Completed"}[$__interval]))
or max_over_time(prefect_info_flow_runs_total{cluster_arn=~"$ecs_cluster_arn",container_name!~"aws-guardduty-agent.*",state_name!~"Completed|Cancelled|Crashed|Failed|Pending|Scheduled"}[$__interval])
)
```

Grafana state_name mapping: 
<img width="592" height="600" alt="Image" src="https://github.com/user-attachments/assets/337f9c14-c204-4a78-b836-457146cdb39a" />

### Application name

Prefect Cloud and prometheus-prefect-exporter

### Application version

prometheus-prefect-exporter:3.0.0

### Explain bug

_No response_

### What you expected to happen?

_No response_

### How to reproduce it?

_No response_

### More info

_No response_

--- a/metrics/api_metric.py
+++ b/metrics/api_metric.py
@@ -44,7 +44,7 @@ class PrefectApiMetric:
         Fetch all items from the endpoint with pagination.
 
         Returns:
-            dict: JSON response containing all items from the endpoint.
+            list: All items from the endpoint, or an empty list on failure.
         """
         endpoint = f"{self.url}/{self.uri}/filter"
         enable_pagination = self.enable_pagination
@@ -54,7 +54,7 @@ class PrefectApiMetric:
 
         # Run the loop until the current page is empty
         while True:
-            resp = requests.Response()
+            resp = None
 
             for retry in range(self.max_retries):
                 data = {
@@ -66,13 +66,17 @@ class PrefectApiMetric:
                 try:
                     resp = requests.post(endpoint, headers=self.headers, json=data)
                     resp.raise_for_status()
-                except requests.exceptions.HTTPError as err:
-                    self.logger.error(err)
-                    if retry >= self.max_retries - 1:
-                        time.sleep(1)
-                        raise SystemExit(err)
-                else:
                     break
+                except requests.exceptions.RequestException as err:
+                    self.logger.error(err)
+                    if retry < self.max_retries - 1:
+                        time.sleep(2**retry)
+                    else:
+                        self.logger.error(
+                            "Max retries reached for %s, returning partial results",
+                            endpoint,
+                        )
+                        return all_items
 
             curr_page_items = resp.json()
 

--- a/metrics/healthz.py
+++ b/metrics/healthz.py
@@ -42,10 +42,10 @@ class PrefectHealthz:
                 self.logger.info(
                     f"Prefect health check: {resp.status_code} - {resp.reason}"
                 )
-            except requests.exceptions.HTTPError as err:
-                self.logger.error(err)
-                if retry >= self.max_retries - 1:
-                    time.sleep(1)
-                    raise SystemExit(err)
-            else:
                 break
+            except requests.exceptions.RequestException as err:
+                self.logger.error(err)
+                if retry < self.max_retries - 1:
+                    time.sleep(2**retry)
+                else:
+                    raise SystemExit(err)

--- a/metrics/metrics.py
+++ b/metrics/metrics.py
@@ -4,7 +4,7 @@ from datetime import datetime, timezone
 
 import requests
 from prefect.client.schemas.objects import CsrfToken
-from prometheus_client.core import CounterMetricFamily, GaugeMetricFamily
+from prometheus_client.core import GaugeMetricFamily
 
 from metrics.deployments import PrefectDeployments
 from metrics.flow_runs import PrefectFlowRuns
@@ -59,8 +59,19 @@ class PrefectMetrics(object):
 
     def collect(self):
         """
-        Get and set Prefect work queues metrics.
+        Collect all Prefect metrics for a single Prometheus scrape.
 
+        On failure, logs the error and yields no metrics. The exporter stays
+        alive so subsequent scrapes can succeed.
+        """
+        try:
+            yield from self._collect_metrics()
+        except Exception:
+            self.logger.exception("Failed to collect metrics, skipping this scrape")
+
+    def _collect_metrics(self):
+        """
+        Internal method that performs the actual metric collection.
         """
         ##
         # PREFECT GET CSRF TOKEN IF ENABLED
@@ -245,7 +256,7 @@ class PrefectMetrics(object):
         yield prefect_flow_runs
 
         # prefect_flow_runs_total_run_time metric
-        prefect_flow_runs_total_run_time = CounterMetricFamily(
+        prefect_flow_runs_total_run_time = GaugeMetricFamily(
             "prefect_flow_runs_total_run_time",
             "Prefect flow-run total run time in seconds",
             labels=["flow_name"],
@@ -289,7 +300,7 @@ class PrefectMetrics(object):
         yield prefect_flow_runs_total_run_time
 
         # prefect_info_flow_runs metric
-        prefect_info_flow_runs = CounterMetricFamily(
+        prefect_info_flow_runs = GaugeMetricFamily(
             "prefect_info_flow_runs",
             "Prefect flow runs info",
             labels=[
@@ -440,21 +451,19 @@ class PrefectMetrics(object):
         """
         Pull CSRF Token from CSRF Endpoint.
 
+        Raises:
+            requests.exceptions.RequestException: If all retries are exhausted.
         """
-        csrf_token = requests.Response()
+        endpoint = f"{self.url}/csrf-token?client={self.client_id}"
 
         for retry in range(self.max_retries):
             try:
-                csrf_token = requests.get(
-                    f"{self.url}/csrf-token?client={self.client_id}",
-                    headers=self.headers,
-                )
-            except requests.exceptions.HTTPError as err:
+                resp = requests.get(endpoint, headers=self.headers)
+                resp.raise_for_status()
+                return CsrfToken.model_validate(resp.json())
+            except requests.exceptions.RequestException as err:
                 self.logger.error(err)
-                if retry >= self.max_retries - 1:
-                    time.sleep(1)
-                    raise SystemExit(err)
-            else:
-                break
-
-        return CsrfToken.model_validate(csrf_token.json())
+                if retry < self.max_retries - 1:
+                    time.sleep(2**retry)
+                else:
+                    raise

--- a/metrics/work_queues.py
+++ b/metrics/work_queues.py
@@ -67,23 +67,24 @@ class PrefectWorkQueues(PrefectApiMetric):
             work_queue_id (uuid.UUID): The UUID of the work queue.
 
         Returns:
-            dict: JSON response containing work queue status information.
+            dict: JSON response containing work queue status information,
+                  or an empty dict on failure.
 
         """
         endpoint = f"{self.url}/{self.uri}/{work_queue_id}/status"
-        resp = requests.Response()
 
         for retry in range(self.max_retries):
             try:
                 resp = requests.get(endpoint, headers=self.headers)
                 resp.raise_for_status()
-
-            except requests.exceptions.HTTPError as err:
+                return resp.json()
+            except requests.exceptions.RequestException as err:
                 self.logger.error(err)
-                if retry >= self.max_retries - 1:
-                    time.sleep(1)
-                    raise SystemExit(err)
-            else:
-                break
-
-        return resp.json()
+                if retry < self.max_retries - 1:
+                    time.sleep(2**retry)
+                else:
+                    self.logger.error(
+                        "Max retries reached for %s, returning empty status",
+                        endpoint,
+                    )
+                    return {}

